R/get_stan_data.R

```diff
@@ -213,45 +334,54 @@ get_stan_data <- function(input_count_data,
-    sigma_ww_site_prior_mean_sd = sigma_ww_site_prior_mean_sd,
-    sigma_ww_site_prior_sd_mean = sigma_ww_site_prior_sd_mean,
-    sigma_ww_site_prior_sd_sd = sigma_ww_site_prior_sd_sd,
-    eta_sd_sd = eta_sd_sd,
-    sigma_i0_prior_mode = sigma_i0_prior_mode,
-    sigma_i0_prior_sd = sigma_i0_prior_sd,
-    p_hosp_prior_mean = p_hosp_mean,
-    p_hosp_sd_logit = p_hosp_sd_logit,
-    p_hosp_w_sd_sd = p_hosp_w_sd_sd,
-    ww_site_mod_sd_sd = ww_site_mod_sd_sd,
-    inf_feedback_prior_logmean = infection_feedback_prior_logmean,
-    inf_feedback_prior_logsd = infection_feedback_prior_logsd,
-    sigma_rt_prior = sigma_rt_prior,
-    log_phi_g_prior_mean = log_phi_g_prior_mean,
-    log_phi_g_prior_sd = log_phi_g_prior_sd,
+    autoreg_rt_a = params$autoreg_rt_a,
+    autoreg_rt_b = params$autoreg_rt_b,
+    autoreg_rt_site_a = params$autoreg_rt_site_a,
+    autoreg_rt_site_b = params$autoreg_rt_site_b,
+    autoreg_p_hosp_a = params$autoreg_p_hosp_a,
+    autoreg_p_hosp_b = params$autoreg_p_hosp_b,
+    inv_sqrt_phi_prior_mean = params$inv_sqrt_phi_prior_mean,
+    inv_sqrt_phi_prior_sd = params$inv_sqrt_phi_prior_sd,
+    r_prior_mean = params$r_prior_mean,
+    r_prior_sd = params$r_prior_sd,
+    log10_g_prior_mean = params$log10_g_prior_mean,
+    log10_g_prior_sd = params$log10_g_prior_sd,
+    i0_over_n_prior_a = 1 + params$i0_certainty * (i0 / pop),
+    i0_over_n_prior_b = 1 + params$i0_certainty * (1 - (i0 / pop)),
+    wday_effect_prior_mean = params$wday_effect_prior_mean,
+    wday_effect_prior_sd = params$wday_effect_prior_sd,
+    initial_growth_prior_mean = params$initial_growth_prior_mean,
+    initial_growth_prior_sd = params$initial_growth_prior_sd,
+    sigma_ww_site_prior_mean_mean = params$sigma_ww_site_prior_mean_mean,
+    sigma_ww_site_prior_mean_sd = params$sigma_ww_site_prior_mean_sd,
+    sigma_ww_site_prior_sd_mean = params$sigma_ww_site_prior_sd_mean,
+    sigma_ww_site_prior_sd_sd = params$sigma_ww_site_prior_sd_sd,
+    eta_sd_sd = params$eta_sd_sd,
+    sigma_i0_prior_mode = params$sigma_i0_prior_mode,
+    sigma_i0_prior_sd = params$sigma_i0_prior_sd,
+    p_hosp_prior_mean = params$p_hosp_mean,
+    p_hosp_sd_logit = params$p_hosp_sd_logit,
+    p_hosp_w_sd_sd = params$p_hosp_w_sd_sd,
+    ww_site_mod_sd_sd = params$ww_site_mod_sd_sd,
+    inf_feedback_prior_logmean = params$infection_feedback_prior_logmean,
+    inf_feedback_prior_logsd = params$infection_feedback_prior_logsd,
+    sigma_rt_prior = params$sigma_rt_prior,
+    log_phi_g_prior_mean = params$log_phi_g_prior_mean,
+    log_phi_g_prior_sd = params$log_phi_g_prior_sd,
     ww_sampled_sites = ww_indices$ww_sampled_sites,
     lab_site_to_site_map = ww_indices$lab_site_to_site_map
   )
 
-  return(data_renewal)
+  input_data <- list(
+    input_ww_data = ww_data,
+    input_count_data = count_data
+  )
+  input_data_and_args <- list(
+    input_data = input_data,
+    stan_args = data_renewal
```

Comment:
In this function, I generate the table of count data and wastewater data that is immediately passed to stan. My thought was that this is the data we want to return to the user in the `wwinference()` wrapper model function as is described in #49, and this is just the place where it is generated. Otherwise, there is no reason to have this nested list.

Reply:
I see, thanks. If that's what you want, I would suggest splitting the activity in this `get_stan_data()` function:
- one function that creates the table currently returned as `input_data_and_args$input_data$input_ww_data`
- a second function that creates the table currently returned as `input_data_and_args$input_data$input_count_data
- A third function that takes those two things plus `generation_interval`, `inf_to_count_delay`, `infection_feedback_pmf`, and `params` as arguments, and creates a properly formatted data list to pass to `sample()`
I would then call those three functions in that order within `wwinference()`

Reply:
Ok I did my best to refactor this, I think one of the issues is that the wastewater data is sort of conditioned on the hospital admissions data (so had to change some things so that we're passing around the first and last date and stuff). Introduced a bit of redundancy that I am not sure is necessary/ideal.